skfp/fingerprints/base.py

```diff
@@ -95,3 +95,16 @@ def _validate_input(self, X: List, smiles_only: bool = False) -> List[Mol]:
 
         X = [MolFromSmiles(x) if isinstance(x, str) else x for x in X]
         return X
+
+    def _hash_fingerprint_bits(self, X: List) -> Union[np.ndarray, csr_array]:
+        shape = (len(X), self.fp_size)
```

Comment:
Maybe it would be worth to create a self.fp_size field in the base class and initialize as 0? Just for the sake of consistency as so far, the base class only referenced its own member variables.

Reply:
This implicitly assumes that inheriting class has `fp_size` attribute, that's true. But I don't think this is avoidable, since we want all fingerprints to inherit from FingerprintTransformer, and all those that need hashing will have `fp_size`.

I will add an explicit check for those attributes, to avoid problems.